libs/core/src/lib/upload-collection/upload-collection-item.directive.ts

```diff
@@ -48,28 +60,76 @@ export class UploadCollectionItemDirective implements AfterContentInit, OnDestro
     /** @hidden */
     private _subscriptions = new Subscription();
 
+    /** @hidden */
+    fileNameFull: string;
+
+    /** @hidden */
+    titleWidth: number;
+
+    /** @hidden used to compare to the current width to know whether to collapse or expand title */
+    previousContainerWidth: number;
+
+    /** @hidden */
+    containerWidth: number;
+
     /** @hidden */
     ngAfterContentInit(): void {
-        this._titleDirective.elRef.nativeElement.innerHTML = this.fileName + '.' + this.extension;
+        this.fileNameFull = this.fileName + '.' + this.extension;
+        this._titleDirective.elRef.nativeElement.innerHTML = this.fileNameFull;
         this._handleDeleteClickedSubscription();
         this._handleOkClickedSubscription();
         this._handleEditClickedSubscription();
         this._handleFormItemInputChangedSubscription();
     }
 
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this.onResize();
+    }
+
     /** @hidden */
     ngOnDestroy(): void {
         this._subscriptions.unsubscribe();
     }
 
+    constructor(public elementRef: ElementRef, private _cdRef: ChangeDetectorRef) {}
+
+    /** @hidden */
+    @HostListener('window:resize', [])
+    onResize(): void {
+        if (!this.elementRef.nativeElement.parentElement) {
+            return;
+        }
+        this.titleWidth = this.getTitleWidth();
+        this.containerWidth = this.getContainerWidth();
+
+        // if first load and no previous container width, or if container boundary is resized to smaller than before
+        if (!this.previousContainerWidth || this.containerWidth < this.previousContainerWidth) {
+            // and the title extends past the container
+            if (this.titleWidth * 1.05 >= this.containerWidth) {
```

Comment:
Why not just `if (this.titleWidth >= this.containerWidth)`

Reply:
explained above, i find that the button-group shifts when the text is nearly identical to the container size, so added some wiggle room.